app/src/main/java/ch/epfl/qedit/view/util/ListEditView.java

```diff
@@ -144,9 +142,12 @@ public void onClick(View v) {
      *
      * @param <T> the type of the underlying items
      */
-    public static class Adapter<T> extends RecyclerView.Adapter<ItemHolder> {
+    public static class Adapter<T, E extends SearchablePair<T>>
+            extends RecyclerView.Adapter<ItemHolder> implements Filterable {
 
-        private final List<T> items;
+        private final E e;
```

Comment:
I think that if we want to keep the `ListEditView` generic and reusable, it is a bad idea to add fields that only serve the purpose of the online activity. I am not saying you shouldn't do it, but really think about another way before doing that. Since other activities use the `ListEditView`, introducing changes might break things in another place...

Reply:
What I would recommend is that you subclass `ListEditView` to add the functionality that you need, so that it doesn't have an impact on the rest of the app.